From 176d79ea889849b1287dd773c5cc1a19b6ae6675 Mon Sep 17 00:00:00 2001
From: Lars Kellogg-Stedman <lars@redhat.com>
Date: Tue, 3 Jun 2014 12:44:06 -0400
Subject: [PATCH 1/2] correct resource dependencies for fedora/redhat

This patches makes Firewall resources depend upon the iptables-services
package.  Without this dependency, firewall rules will be instantiated
in the running configuration but may not persist correctly because the
necessary command is not (yet) available.

Signed-off-by: Gael Chamoulaud <gchamoul@redhat.com>
---
 manifests/linux/redhat.pp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/manifests/linux/redhat.pp b/manifests/linux/redhat.pp
index a4c00b6..08965b8 100644
--- a/manifests/linux/redhat.pp
+++ b/manifests/linux/redhat.pp
@@ -23,13 +23,17 @@ class firewall::linux::redhat (
   if $::operatingsystem == RedHat and $::operatingsystemrelease >= 7 {
     package { 'iptables-services':
       ensure => present,
+      before => Service['iptables'],
     }
+    Package['iptables-services'] -> Firewall <||>
   }
 
   if $::operatingsystem == Fedora and $::operatingsystemrelease >= 15 {
     package { 'iptables-services':
       ensure => present,
+      before => Service['iptables'],
     }
+    Package['iptables-services'] -> Firewall <||>
   }
 
   service { 'iptables':
-- 
1.9.3


From 2e6885fc8e502beeb903edcda4b1fa6a7a5f1326 Mon Sep 17 00:00:00 2001
From: Gael Chamoulaud <gchamoul@redhat.com>
Date: Wed, 9 Jul 2014 12:54:53 +0200
Subject: [PATCH 2/2] Fixes CentOS 7 new version numbers

This patches fixes the comparison issue with the new CentOS 7 version numbers.

Signed-off-by: Gael Chamoulaud <gchamoul@redhat.com>
---
 manifests/linux/redhat.pp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/manifests/linux/redhat.pp b/manifests/linux/redhat.pp
index 08965b8..427f179 100644
--- a/manifests/linux/redhat.pp
+++ b/manifests/linux/redhat.pp
@@ -17,10 +17,12 @@ class firewall::linux::redhat (
   $enable = true
 ) {
 
+  $el_release = ['RedHat', 'CentOS']
+
   # RHEL 7 and later and Fedora 15 and later require the iptables-services
   # package, which provides the /usr/libexec/iptables/iptables.init used by
   # lib/puppet/util/firewall.rb.
-  if $::operatingsystem == RedHat and $::operatingsystemrelease >= 7 {
+  if $::operatingsystem in $el_release and $::operatingsystemmajrelease >= 7 {
     package { 'iptables-services':
       ensure => present,
       before => Service['iptables'],
-- 
1.9.3


